src/idl_gen_cpp.cpp

```diff
@@ -100,6 +100,13 @@ class CppGenerator : public BaseGenerator {
 
     assert(!cur_name_space_);
 
+    code_ += "#if defined(_MSC_VER)";
```

Comment:
rather than generate code for this every time, stick it in `flatbuffers.h` (and call it `FLATBUFFERS_NOEXCEPT` to avoid clashes).